rasa/core/policies/memoization.py

```diff
@@ -236,6 +237,17 @@ def predict_action_probabilities(
 
         return result
 
+    def format_tracker_states(self, states):
```

Comment:
could you please also add the rest of the information like active loops, slots and entities?

Reply:
I think it is better to move this method to `Policy` superclass, then you can call `self. format_tracker_states` in all other policies

Reply:
Thanks for your review! We pushed updates, now ` format_tracker_states` is into  `policy.py`. But We not sure about where the rest of the information should be displayed in logging. Would you mind telling us where do you think is a good place for it?

Reply:
the state lists might contain keys like `USER, PREVIOUS_ACTION, ACTIVE_LOOP, SLOTS` (these are constant names for strings defined in `rasa.core.constants`).
`USER` can contain keys defined here: https://github.com/RasaHQ/rasa/blob/005f8b1e4742efca12ee56194bce4f1985e73555/rasa/shared/core/events.py#L390
`PREVIOUS_ACTION` can contain keys defined here: https://github.com/RasaHQ/rasa/blob/005f8b1e4742efca12ee56194bce4f1985e73555/rasa/shared/core/events.py#L1167


If you want to reformat `states` dict, then all of possible states should be outputted and not only

Reply:
@Ghostvv Sorry we took so long!!! We not sure our solution is the most elegant one but was the one that works. We will be glad if could take a look.

Reply:
We also developed an alternate solution, focusing on specific dict keys to build the final message. We could, also, implement this same approach on a generic form, without using the specific names of the keys but iterating through the dict. Here's how we pictured it could work:

```python
def format_tracker_states(self, states: List[Dict]) -> Text:
        """Format tracker states to human readable format on debug log

        Args: 
            states: list of tracker states dicts
        Returns:
            the string of the states with user intents and actions
        """
        # empty string to insert line break before first state
        formatted_states = [""]
        if states:
            for index, state in enumerate(states):
                state_messages = []
                if state:
                    if USER in state:
                        if TEXT in state[USER]:
                            state_messages.append(
                                f"user text: {str(state[USER][TEXT])}"
                            )
                        if INTENT in state[USER]:
                            state_messages.append(
                                f"user intent: {str(state[USER][INTENT])}"
                            )
                        if ENTITIES in state[USER]:
                            state_messages.append(
                                f"user entities: {str(state[USER][ENTITIES])}"
                            )
                    if PREVIOUS_ACTION in state:
                        if ACTION_NAME in state[PREVIOUS_ACTION]:
                            state_messages.append(
                                f"previous action name: {str(state[PREVIOUS_ACTION][ACTION_NAME])}"
                            )
                        if ACTION_TEXT in state[PREVIOUS_ACTION]:
                            state_messages.append(
                                f"previous action text: {str(state[PREVIOUS_ACTION][ACTION_TEXT])}"
                            )
                    if ACTIVE_LOOP in state:
                        state_messages.append(
                            f"active loop: {str(state[ACTIVE_LOOP])}"
                        )
                    if SLOTS in state:
                        state_messages.append(
                            f"slots: {str(state[SLOTS])}"
                        )
                    state_message_formatted = " | ".join(state_messages)
                    state_formatted = f"[state {str(index)}] {state_message_formatted}" 
                    formatted_states.append(state_formatted)

        return "\n".join(formatted_states)
```

Reply:
the explicit approach looks good to me, we also use similar explicit keys in single state featurizer. Not sure if key agnostic for over dict will make it simpler since you need to create key specific strings anyhow